lib/spack/spack/build_environment.py

```diff
@@ -750,6 +750,9 @@ def setup_package(pkg, dirty, context='build'):
     elif context == 'test':
         import spack.user_environment as uenv  # avoid circular import
         env.extend(uenv.environment_modifications_for_spec(pkg.spec))
+        env.extend(
```

Comment:
Hmm.  This fix was integrated in #20298 .  Unfortunately, it isn't quite enough in some cases (#20326 ).